common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -51,3 +54,105 @@ def perform_destroy(self, instance):
             )
         if save_model:
             instance.save()
+
+
+class EntitlementEnrollmentViewSet(viewsets.GenericViewSet):
+    authentication_classes = (JwtAuthentication, SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated,)
+    queryset = CourseEntitlement.objects.all()
+    serializer_class = CourseEntitlementSerializer
+
+    def _enroll_entitlement(self, entitlement, course_session_key, user):
+        enrollment = CourseEnrollment.enroll(
+            user=user,
+            course_key=course_session_key,
+            mode=entitlement.mode,
+        )
+
+        CourseEntitlement.set_enrollment(entitlement, enrollment)
+
+    def _unenroll_entitlement(self, entitlement, course_session_key, user):
+        CourseEnrollment.unenroll(user, course_session_key, skip_refund=True)
+        CourseEntitlement.set_enrollment(entitlement, None)
+
+    def create(self, request, uuid):
+        course_session_id = request.data.get('course_run_id', None)
+
+        if not course_session_id:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data="The Course Run ID was not provided."
```

Comment:
Should this text be translated?

Reply:
Note: when translating these we may need to escape the translations differently since they'll be provided via the JSON? response.

Reply:
We do not need to translate strings meant for Devs in the API.  This is not meant for display.